internal/website/content/howto/pubsub/publish.md

```diff
@@ -7,40 +7,40 @@ toc: true
 
 Publishing a message to a topic with the Go CDK takes two steps:
 
-1. Open the topic with the Pub/Sub provider of your choice (once per topic).
-2. Send messages on the topic.
+1. [Open the topic][] with the Pub/Sub provider of your choice (once per topic).
+2. [Send messages][] on the topic.
 
-<!--more-->
+[Open the topic]: {{< ref "#opening" >}}
+[Send messages]: {{< ref "#sending" >}}
 
-The second step is the same across all providers because the first step
-creates a value of the portable [`*pubsub.Topic`][] type. Publishing looks
-like this:
-
-{{< goexample src="gocloud.dev/pubsub.ExampleTopic_Send" imports="0" >}}
-
-Note that the [semantics of message delivery][] can vary by provider.
-
-The rest of this guide will discuss how to accomplish the first step: opening
-a topic for your chosen Pub/Sub provider.
-
-[`*pubsub.Topic`]: https://godoc.org/gocloud.dev/pubsub#Topic
-[semantics of message delivery]: https://godoc.org/gocloud.dev/pubsub#hdr-At_most_once_and_At_least_once_Delivery
-
-## Constructors versus URL openers
+## Opening a Topic {#opening}
 
 The easiest way to open a topic is using [`pubsub.OpenTopic`][] and a URL
 pointing to the topic, making sure you ["blank import"][] the driver package to
 link it in. See [Concepts: URLs][] for more details. If you need fine-grained
 control over the connection settings, you can call the constructor function in
-the driver package directly (like `gcppubsub.OpenTopic`). This guide will show
-how to use both forms for each pub/sub provider.
+the driver package directly (like `gcppubsub.OpenTopic`). [This guide][] will
+show how to use both forms for each pub/sub provider.
 
+[This guide]: {{< ref "#services" >}}
 [`pubsub.OpenTopic`]:
 https://godoc.org/gocloud.dev/pubsub#OpenTopic
 ["blank import"]: https://golang.org/doc/effective_go.html#blank_import
 [Concepts: URLs]: {{< ref "/concepts/urls.md" >}}
 
-## Amazon Simple Notification Service {#sns}
+## Sending messages on a Topic {#sending}
+
+Sending a message on a Topic looks like this:
+
+{{< goexample src="gocloud.dev/pubsub.ExampleTopic_Send" imports="0" >}}
+
+Note that the [semantics of message delivery][] can vary by provider.
+
+[semantics of message delivery]: https://godoc.org/gocloud.dev/pubsub#hdr-At_most_once_and_At_least_once_Delivery
+
+## Supported Pub/Sub Services {#services}
+
+### AWS Simple Notification Service {#sns}
```

Comment:
According to Google the long name with all words spelled out starts with Amazon, not AWS. The abbreviated version is used with as both Amazon SNS and AWS SNS.

Reply:
Will change to Amazon here